core/src/main/scala/caliban/wrappers/CostEstimation.scala

```diff
@@ -201,13 +203,15 @@ object CostEstimation {
         process: Document => ZIO[R1, ValidationError, ExecutionRequest]
       ): Document => ZIO[R1, ValidationError, ExecutionRequest] =
         (doc: Document) =>
-          for {
-            req  <- process(doc)
-            cost <- computeCostZIO(req.field)(f)
-            _    <- ZIO.when(cost > maxCost)(
-                      ZIO.fail(ValidationError(s"Query costs too much: $cost. Max cost: $maxCost.", ""))
-                    )
-          } yield req
+          process(doc).tap { req =>
+            ZIO.unlessZIO(Validator.skipQueryValidationRef.get) {
```

Comment:
I'm wondering if this needs another parameter to allow the user to bypass the validation skip here. Since it is possible that the `Double` is returning some aggregate cost value. So the user may want this to apply to all queries regardless of if they are persisted

Reply:
That's a good point. I added the `skipForPersistedQueries: Boolean = false` argument to the `maxCostXX` methods. I used `false` as the default value so that we don't affect the behaviour for existing / unaware users